examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -57,6 +61,119 @@ void check_has_no_field(const T& document, const char* field, int example_no) {
+        "MONGOCXX_TEST_AZURE_TENANT_ID",
+        "MONGOCXX_TEST_AZURE_CLIENT_ID",
+        "MONGOCXX_TEST_AZURE_CLIENT_SECRET",
+        "MONGOCXX_TEST_CSFLE_TLS_CA_FILE",
+        "MONGOCXX_TEST_CSFLE_TLS_CERTIFICATE_KEY_FILE",
+        "MONGOCXX_TEST_GCP_EMAIL",
+        "MONGOCXX_TEST_GCP_PRIVATEKEY",
+    };
+
+    std::ostringstream os;
+    os << "Please set environment variables to enable client side encryption tests:\n";
+    std::copy(std::begin(vars), std::end(vars), std::ostream_iterator<const char*>(os, "\n"));
+
+    if (std::none_of(std::begin(vars), std::end(vars), std::getenv)) {
+        os << "Skipping client side encryption tests.\n";
+        std::cerr << os.str() << std::endl;
+        return false;
+    }
+
+    if (!std::all_of(std::begin(vars), std::end(vars), std::getenv)) {
+        os << "Failing client side encryption tests (some environment variables were not set).\n";
+        std::cerr << os.str() << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+mongocxx::options::client add_test_server_api(mongocxx::options::client opts = {}) {
+    auto api_version = std::getenv("MONGODB_API_VERSION");
+    if (api_version && std::string(api_version).length() > 0) {
+        if (!std::string(api_version).compare("1")) {
+            opts.server_api_opts(
+                mongocxx::options::server_api(mongocxx::options::server_api::version::k_version_1));
+        } else {
+            throw std::logic_error("invalid server API version");
+        }
+    }
+    return opts;
+}
+
+std::string getenv_or_fail(const char* s) {
+    const char* env = std::getenv(s);
+    if (!env) {
+        throw std::runtime_error("missing a required environment variable: " + std::string(s));
+    }
+    return std::string(env);
+}
+
+// Returns a document with credentials for KMS providers.
+// If include_external is true, all KMS providers are set.
+// If include_external is false, only the local provider is set.
+bsoncxx::document::value _make_kms_doc(bool include_external = true) {
+    using bsoncxx::builder::basic::sub_document;
+    using bsoncxx::builder::stream::close_array;
+    using bsoncxx::builder::stream::close_document;
+    using bsoncxx::builder::stream::document;
+    using bsoncxx::builder::stream::finalize;
+    using bsoncxx::builder::stream::open_array;
+    using bsoncxx::builder::stream::open_document;
+
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    const auto kLocalMasterKey =
+        "\x32\x78\x34\x34\x2b\x78\x64\x75\x54\x61\x42\x42\x6b\x59\x31\x36\x45\x72"
+        "\x35\x44\x75\x41\x44\x61\x67\x68\x76\x53\x34\x76\x77\x64\x6b\x67\x38\x74"
+        "\x70\x50\x70\x33\x74\x7a\x36\x67\x56\x30\x31\x41\x31\x43\x77\x62\x44\x39"
+        "\x69\x74\x51\x32\x48\x46\x44\x67\x50\x57\x4f\x70\x38\x65\x4d\x61\x43\x31"
+        "\x4f\x69\x37\x36\x36\x4a\x7a\x58\x5a\x42\x64\x42\x64\x62\x64\x4d\x75\x72"
+        "\x64\x6f\x6e\x4a\x31\x64";
```

Comment:
```suggestion
    const uint8_t kLocalMasterKey[] =
        "\x32\x78\x34\x34\x2b\x78\x64\x75\x54\x61\x42\x42\x6b\x59\x31\x36\x45\x72"
        "\x35\x44\x75\x41\x44\x61\x67\x68\x76\x53\x34\x76\x77\x64\x6b\x67\x38\x74"
        "\x70\x50\x70\x33\x74\x7a\x36\x67\x56\x30\x31\x41\x31\x43\x77\x62\x44\x39"
        "\x69\x74\x51\x32\x48\x46\x44\x67\x50\x57\x4f\x70\x38\x65\x4d\x61\x43\x31"
        "\x4f\x69\x37\x36\x36\x4a\x7a\x58\x5a\x42\x64\x42\x64\x62\x64\x4d\x75\x72"
        "\x64\x6f\x6e\x4a\x31\x64";

    static_assert(sizeof(kLocalMasterKey) - 1u == 96u, "key should be exactly 96 bytes");
```
Avoid unnecessary casts later below.